php/Blockonomics.php

```diff
@@ -42,36 +54,50 @@ public function new_address($api_key, $secret, $reset=false)
 
     public function get_price($currency)
     {
-    	$url = Blockonomics::PRICE_URL. "?currency=$currency";
+        $url = Blockonomics::PRICE_URL. "?currency=$currency";
         $response = $this->get($url);
         return json_decode(wp_remote_retrieve_body($response))->price;
     }
 
-    public function get_xpubs($api_key)
+    public function update_callback($callback_url, $xpub)
     {
-    	$url = Blockonomics::ADDRESS_URL;
-        $response = $this->get($url, $api_key);
+        $url = Blockonomics::SET_CALLBACK_URL;
+        $body = json_encode(array('callback' => $callback_url, 'xpub' => $xpub));
+        $response = $this->post($url, $this->api_key, $body);
         return json_decode(wp_remote_retrieve_body($response));
     }
 
-    public function update_callback($api_key, $callback_url, $xpub)
+    public function get_callbacks()
     {
-    	$url = Blockonomics::SET_CALLBACK_URL;
-    	$body = json_encode(array('callback' => $callback_url, 'xpub' => $xpub));
-    	$response = $this->post($url, $api_key, $body);
-        return json_decode(wp_remote_retrieve_body($response));
+        $url = Blockonomics::GET_CALLBACKS_URL;
+        $response = $this->get($url, $this->api_key);
+        return $response;
     }
 
-    public function get_callbacks($api_key)
+    public function get_temp_api_key($callback_url)
     {
-    	$url = Blockonomics::GET_CALLBACKS_URL;
-    	$response = $this->get($url, $api_key);
-        return $response;
+        $url = Blockonomics::TEMP_API_KEY_URL;
+        $body = json_encode(array('callback' => $callback_url));
+        $response = $this->post($url, '', $body);
+        $responseObj = json_decode(wp_remote_retrieve_body($response));
+        $responseObj->{'response_code'} = wp_remote_retrieve_response_code($response);
+        return $responseObj;
+    }
+
+    public function make_withdraw()
+    {
+        $api_key = get_option("blockonomics_api_key");
+        $temp_api_key = get_option('blockonomics_temp_api_key');
```

Comment:
Do some sanity checks.. if temp_api_key is empty or api_key doesn't exist ... then do nothing